README.md

```diff
@@ -57,39 +57,54 @@ dependencies {
 
 The core of lite-for-jdbc is the Db class. A Db object is intended to be used as a singleton and
 injected as a dependency in repository classes. It requires a DataSource constructor argument,
-and there is a DataSourceFactory to help with that.
+and there is a DataSourceFactoryRegistry to help with that.
 The typical recommendation is to use Hikari, which is configured with reasonable defaults, but you can customize
 it to any DataSource.
 Examples:
 
-Hikari:
+Using DataSourceFactoryRegistry directly:
 
 ```kotlin
-val dataSource = DatasourceFactory(DbConfig(
-    username = "user",
-    password = "password",
-    databaseName = "dbName"
-)).dataSource()
+val config = DbConfig(
+  type = "H2_INMEM",
+  username = "user",
+  password = "password",
+  databaseName = "dbName"
+)
+val datasource = DataSourceFactoryRegistry.dataSource(config)
 
-val db = Db(dataSource)
+val db = Db(datasource)
 ```
 
-H2 in-memory:
+Or you can use the Db constructor that accepts a DbConfig directly. Db will use DataSourceFactoryRegistry under the covers for you.
 
 ```kotlin
-val dataSource = DatasourceFactory(DbConfig(
+val db = Db(DbConfig(
   type = "H2_INMEM",
   username = "user",
   password = "password",
   databaseName = "dbName"
-)).dataSource()
-
-val db = Db(dataSource)
+))
 ```
 
-See DbConfig for a full list of configuration options available.
+See `DbConfig` for a full list of configuration options available.
+
+## Custom Database Types
 
-If another implementation of DataSource is required, you can construct your own as shown below:
+If another implementation of DataSource is required, you can register a custom "Type" to be set on `DbConfig` and build the 
+respective `DataSource` in a `DataSourceFactory` lambda as shown below.
+
+```kotlin
+DataSourceFactoryRegistry.registerDataSourceFactory("custom") {config: DbConfig ->
```

Comment:
`Type` is what naturally came to my mind, and then I hunted around your comments and saw you said "you can register a custom Type to be set..."
```suggestion
DataSourceFactory.addType("custom") {config: DbConfig ->
```